### PR TITLE
[Communication] Add ACS rooms support

### PR DESCRIPTION
**Pull request by mayssamm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Communication

File: src/communication/README.md [hunk 1]
```diff
@@ -146,3 +146,35 @@ az communication chat message receipt list --thread "19:xxxxxx" --skip "5"
 ```
 az communication chat message receipt send --thread "19:xxxxxx" --message-id "1xxxxxxxxxxxx"
 ```
+##### Get-Room #####
+```
+az communication rooms get --room "roomId"
```
Comment (zhoxing-ms) [inline src/communication/README.md:151]: Our command signature usually does not use the plural of nouns
```suggestion
az communication room get --room "roomId"
```

File: src/communication/azext_communication/manual/_help.py [hunk 2]
```diff
@@ -407,3 +407,100 @@
...
 """
+
+helps['communication rooms'] = """
+    type: group
+    short-summary: Commands to interact with Azure Communication Services Rooms gateway.
+"""
+
+helps['communication rooms create'] = """
+    type: command
+    short-summary: "Creates a new room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:418]: Please use the first person voice in the help message
```suggestion
    short-summary: "Create a new room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 3]
```diff
@@ -407,3 +407,100 @@
...
+    short-summary: "Returns attributes of an existing room."
+    examples:
+      - name: Get room
+        text: |-
+               az communication rooms get --room "12345678901234567"
+"""
+
+helps['communication rooms update'] = """
+    type: command
+    short-summary: "Updates attributes of an existing room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:451]:
```suggestion
    short-summary: "Update attributes of an existing room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 4]
```diff
@@ -407,3 +407,100 @@
...
+    examples:
+      - name: Update a room's valid-from and valid-until properties
+        text: |-
+               az communication rooms update --room "12345678901234567" --valid-from "2022-07-14T10:21" --valid-until "2022-08-14T10:21"
+"""
+
+helps['communication rooms delete'] = """
+    type: command
+    short-summary: "Deletes an existing room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:460]:
```suggestion
    short-summary: "Delete an existing room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 5]
```diff
@@ -407,3 +407,100 @@
...
+"""
+
+helps['communication rooms participant'] = """
+    type: group
+    short-summary: Commands to manage participants of an existing room.
+"""
+
+helps['communication rooms participant get'] = """
+    type: command
+    short-summary: "Gets participants of a room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:474]:
```suggestion
    short-summary: "Get participants of a room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 6]
```diff
@@ -407,3 +407,100 @@
...
+    examples:
+      - name: Get room's participants
+        text: |-
+               az communication rooms participant get --room "12345678901234567"
+"""
+
+helps['communication rooms participant add'] = """
+    type: command
+    short-summary: "Adds participants to a room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:483]:
```suggestion
    short-summary: "Add participants to a room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 7]
```diff
@@ -407,3 +407,100 @@
...
+    examples:
+      - name: Add presenter and attendee participants to a room
+        text: |-
+               az communication rooms participant add --room "12345678901234567" --presenter-participants "8:acs:xxxxxx" "8:acs:xxxxxx" --attendee-participants "8:acs:xxxxxx" "8:acs:xxxxxx"
+"""
+
+helps['communication rooms participant update'] = """
+    type: command
+    short-summary: "Replaces the participants of a room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:492]:
```suggestion
    short-summary: "Replace the participants of a room."
```

File: src/communication/azext_communication/manual/_help.py [hunk 8]
```diff
@@ -407,3 +407,100 @@
...
+    examples:
+      - name: Update room participants
+        text: |-
+               az communication rooms participant update --room "12345678901234567" --presenter-participants "8:acs:xxxxxx" "8:acs:xxxxxx" --attendee-participants "8:acs:xxxxxx" "8:acs:xxxxxx"
+"""
+
+helps['communication rooms participant remove'] = """
+    type: command
+    short-summary: "Removes participants from a room."
```
Comment (zhoxing-ms) [inline src/communication/azext_communication/manual/_help.py:501]:
```suggestion
    short-summary: "Remove participants from a room."
```

File: src/communication/README.md (hunk 1 above)
Comment (mayssamm) [reply src/communication/README.md:151]: Thank you, Xing, for the review. I understand, and we have used singular form for other command groups under communication, but in this specific case, the service itself is called rooms and I think it is better to stay with the plural form "rooms" for consistency.

File: src/communication/azext_communication/manual/_help.py (hunk 2 above)
Comment (mayssamm) [reply src/communication/azext_communication/manual/_help.py]: Thank you, fixed.

File: src/communication/azext_communication/manual/_params.py [hunk 9]
```diff
@@ -167,3 +168,77 @@ def _load_chat_message_management(self):
...
+    with self.argument_context('communication rooms create') as c:
+        c.argument('valid_from', options_list=['--valid-from'], type=str,
+                   help='The timestamp from when the room is open for joining, '
+                   'in in ISO8601 format, ex: 2022-07-14T10:21. Optional.')
+        c.argument('valid_until', options_list=['--valid-until'], type=str,
+                   help='The timestamp from when the room can no longer be joined,'
+                   ' in ISO8601 format, ex: 2022-07-14T10:21. Optional.')
+        c.argument('join_policy', options_list=['--join-policy'], type=str,
+                   help='The join policy of the room. '
+                   'Can be InviteOnly or CommunicationServiceUsers. Optional.')
```
[the remark this answers is not in the document]
Comment (mayssamm) [reply src/communication/azext_communication/manual/_params.py]: Fixed, thanks.

File: src/communication/azext_communication/manual/commands.py [hunk 10]
```diff
@@ -69,3 +83,22 @@ def load_command_table(self, _):
...
+def _load_rooms_command_table(self):
+    rooms_arguments = ['connection_string']
+    self.command_group('communication rooms', is_preview=True)
+
+    # room management
+    with self.command_group('communication rooms', client_factory=cf_communication_rooms, is_preview=True) as g:
+        g.communication_custom_command('get', 'communication_rooms_get_room', rooms_arguments)
+        g.communication_custom_command('create', 'communication_rooms_create_room', rooms_arguments)
+        g.communication_custom_command('update', 'communication_rooms_update_room', rooms_arguments)
+        g.communication_custom_command('delete', 'communication_rooms_delete_room', rooms_arguments)
```
[the remark this answers is not in the document]
Comment (mayssamm) [reply src/communication/azext_communication/manual/commands.py]: Good suggestion, done.

File: src/communication/azext_communication/manual/commands.py [hunk 11]
```diff
@@ -69,3 +83,22 @@ def load_command_table(self, _):
...
+
+    # rooms participant management
+    with self.command_group('communication rooms participant', client_factory=cf_communication_rooms, is_preview=True) as g:
+        g.communication_custom_command('get', 'communication_rooms_get_participants', rooms_arguments)
+        g.communication_custom_command('add', 'communication_rooms_add_participants', rooms_arguments)
+        g.communication_custom_command('update', 'communication_rooms_update_participants', rooms_arguments)
+        g.communication_custom_command('remove', 'communication_rooms_remove_participants', rooms_arguments)
```
[the remark this answers is not in the document]
Comment (mayssamm) [reply src/communication/azext_communication/manual/commands.py]: Thanks, added confirmation for all delete/remove commands under communication.

**Comment by yonzhan** [comment]
Communication

File: src/communication/README.md (hunk 1 above)
Comment (mayssamm) [reply src/communication/README.md:151]: I put a comment here to keep this conversation open and gather more feedback.

Comment (radubulboaca) [reply src/communication/README.md:151]: Thanks for raising this @zhoxing-ms! In this particular case, the use of the plural form may be justified. 

For better or worse, the name of the service that is using the plural form. The same plural form is used in external documentation, APIs, SDKs and more. Here is the [quick start documentation](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/rooms/get-started-rooms) for reference. 

In this case, as service owners, we would prefer to have consistent naming across various mediums, including the CLI, in the interest of making it easy for developers to form associations.

CC: @mayssamm

Comment (zhoxing-ms) [reply src/communication/README.md:151]: OK, I can accept this reason. Let's see the opinions from assignees @kairu-ms @jsntcy

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**